src/vivarium_csu_ltbi/components/treatment.py

```diff
@@ -0,0 +1,165 @@
+        six_h_adherence_data = builder.data.load("six_h.adherence")
+        self.adherence = builder.lookup.build_table(pd.concat([three_hp_adherence_data, six_h_adherence_data], axis=0),
+                                                    parameter_columns=['age', 'year'],
+                                                    key_columns=['sex', 'treatment_type'],
+                                                    value_columns=['value'])
+
+        six_h_coverage_data = builder.data.load("six_h.coverage.proportion")
+        self.six_h_with_hiv, self.six_h_under_five_hhtb = self.setup_coverage_tables(builder, six_h_coverage_data)
+        three_hp_coverage_data = builder.data.load("three_hp.coverage.proportion")
+        self.three_hp_with_hiv, self.three_hp_under_five_hhtb = self.setup_coverage_tables(builder,
+                                                                                           three_hp_coverage_data)
+
+        self.coverage = builder.value.register_value_producer('ltbi_treatment.coverage', source=self.get_coverage,
+                                                              preferred_post_processor=self.enforce_not_eligible)
+
+        self._treatment_status = pd.Series()
+        self.treatment_status = builder.value.register_value_producer('treatment_status.category',
+                                                                      source=lambda index:
+                                                                      self._treatment_status[index])
+
+        self.columns_created = ['treatment_date', 'treatment_type', 'adherence_propensity']
+        builder.population.initializes_simulants(self.on_initialize_simulants,
+                                                 requires_columns=[],
+                                                 creates_columns=self.columns_created)
+
+        self.disease_state_column = "tuberculosis_and_hiv"
+        self.population_view = builder.population.get_view([self.disease_state_column, 'age', 'sex', 'alive'] +
+                                                           self.columns_created,
+                                                           query="alive == 'alive' & treatment_type == 'untreated'")
+
+        builder.event.register_listener('time_step__prepare', self.on_time_step_prepare)
+
+    def on_initialize_simulants(self, pop_data):
+
+        self._treatment_status = self._treatment_status.append(pd.Series('untreated', index=pop_data.index))
+        initialized = pd.DataFrame({'treatment_date': pd.NaT,
+                                    'treatment_type': 'untreated',
+                                    'adherence_propensity': self.adherence_stream.get_draw(pop_data.index)},
+                                   index=pop_data.index)
+        self.population_view.update(initialized)
+
+    def on_time_step_prepare(self, event):
+        pop = self.population_view.get(event.index)
+        assert '6H' not in pop.treatment_type
```

Comment:
This should fail after people start treatment

Reply:
oh this should have been removed. But it should always be true, because the pop view filters out treated, that's what I was checking